postgres.go

```diff
@@ -186,10 +186,8 @@ func (dialector Dialector) DataTypeOf(field *schema.Field) string {
 		}
 		return "text"
 	case schema.Time:
-		if field.Precision > 0 {
-			return fmt.Sprintf("timestamptz(%d)", field.Precision)
-		}
-		return "timestamptz"
+		// in postgresql, Precision == 0, This means being accurate to the seconds.
```

Comment:
the default value of precision is 0, that why we ignored the zero value.

Reply:
but this changed the default behavior and removed the default precision?

Maybe we need to check field's tag by using `field.TagSettings["PRECISION"]`  if the precision's value is 0?

Reply:
The change here doesn't seem to make sense, because there was a problem here.
`field.DataType` is set by user, when we using `gorm:"type:"timestamptz"`, it's not equal `schema.Time`, and when we using `gorm:"type:"time"`, it's means `time of day (no date)`.

https://www.postgresql.org/docs/14/datatype-datetime.html